### PR TITLE
feat: UI changes for batches github app integration

### PR DESCRIPTION
**Pull request by bahrmichael**
This PR adds the UI changes for SRCH-689

Reviewers, please include in your comments if you have a blocking change, or if it's something that can be done in a follow-up PR.

## Test plan

- Manual testing: I can create and delete github apps. I can see the correct one in each place, and there are no problems with PATs.

### CODE REVIEW

File: client/web/src/enterprise/batches/settings/AppDetailsControls.tsx
Comment (bahrmichael) [inline client/web/src/enterprise/batches/settings/AppDetailsControls.tsx:1]: This code was mostly pulled from `client/web/src/enterprise/batches/settings/CommitSigningIntegrationNode.tsx`.

**Comment by bahrmichael** [comment]
Screenshot 2024-07-08 at 12 00 32

We don't have the page for user github apps yet. To meet the deadline, I would rather disable/remove the edit button, and work on that after merging this PR.

File: cmd/frontend/graphqlbackend/batches.graphql [hunk 1]
```diff
@@ -3495,6 +3495,8 @@ type BatchChangesCodeHost {
     """
     commitSigningConfiguration: CommitSigningConfiguration
 
+    gitHubApp: GitHubApp
```
Comment (bahrmichael) [inline cmd/frontend/graphqlbackend/batches.graphql:3498]: We could move this field into the BatchChangesCredential type.

File: client/web/src/enterprise/batches/settings/AppDetailsControls.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+import styles from './CommitSigningIntegrationNode.module.scss'
+
+interface AppDetailsControlsProps {
+    baseURL: string
+    config: Pick<GitHubAppByIDFields, 'id' | 'name' | 'appURL' | 'logo' | 'appID'>
+    refetch: () => void
+}
+
+export const AppDetailsControls: React.FunctionComponent<AppDetailsControlsProps> = ({ baseURL, config, refetch }) => {
```
Comment (BolajiOlajide) [inline client/web/src/enterprise/batches/settings/AppDetailsControls.tsx:38]: Can we name this file and component `GitHubApp...` so it's clear the app being referred to is `GitHubApps`

File: client/web/src/enterprise/batches/settings/CodeHostConnections.tsx [hunk 3]
```diff
@@ -46,6 +53,7 @@ export const UserCodeHostConnections: React.FunctionComponent<
         connectionResult={useUserBatchChangesCodeHostConnection(user.id)}
         headerLine={headerLine}
         user={user}
+        kind={GitHubAppKind.SITE_CREDENTIAL}
```
Comment (BolajiOlajide) [inline client/web/src/enterprise/batches/settings/CodeHostConnections.tsx:56]: This looks like the wrong `kind` was set here. Also can we rename this prop to `GitHubAppKind` so it's clear what `kind` we are referring to here.

File: cmd/frontend/graphqlbackend/batches.graphql (hunk 1 above)
Comment (BolajiOlajide) [reply cmd/frontend/graphqlbackend/batches.graphql:3501]: Yes, it should be on the `BatchChangesCredential` type.

File: client/web/src/enterprise/batches/settings/CodeHostConnections.tsx (hunk 3 above)
Comment (bahrmichael) [reply client/web/src/enterprise/batches/settings/CodeHostConnections.tsx:56]: You're right, this should've been user credential. Fix (also for renaming) incoming.

File: client/web/src/enterprise/batches/settings/AppDetailsControls.tsx (hunk 2 above)
Comment (bahrmichael) [reply client/web/src/enterprise/batches/settings/AppDetailsControls.tsx:38]: Yup!

File: cmd/frontend/graphqlbackend/batches.graphql (hunk 1 above)
Comment (bahrmichael) [reply cmd/frontend/graphqlbackend/batches.graphql]: Updated!

File: cmd/frontend/graphqlbackend/batches.graphql [hunk 4]
```diff
@@ -3562,6 +3562,11 @@ type BatchChangesCredential implements Node {
     Whether the credential is tied to a GitHub App installation.
     """
     isGitHubApp: Boolean!
```
Comment (bahrmichael) [inline cmd/frontend/graphqlbackend/batches.graphql:3564]: I feel like the field isGitHubApp could be removed when the client can check if the gitHubApp is null or not. If possible, I'd drop this field in a follow-up PR. It's pretty new, so no integration should be relying on this.

Comment (BolajiOlajide) [reply cmd/frontend/graphqlbackend/batches.graphql:3564]: Yes, let's do that in a follow up PR.

**Review by BolajiOlajide — APPROVED** [review]

File: cmd/frontend/graphqlbackend/batches.graphql (hunk 4 above)
Comment (bahrmichael) [reply cmd/frontend/graphqlbackend/batches.graphql:3564]: 👍 I created SRCH-704 to track that

**Pull request merged**